Tests/Adyen Tests/Analytics/AnalyticsProviderTests.swift

```diff
@@ -263,15 +263,56 @@ class AnalyticsProviderTests: XCTestCase {
         sut.sendInitialAnalytics(with: .components(type: .achDirectDebit), additionalFields: nil)
         wait(for: .milliseconds(100))
         
-        sut.add(info: AnalyticsEventInfo(component: "card", type: .rendered))
+        let infoEvent = AnalyticsEventInfo(component: "card", type: .rendered)
+        sut.add(info: infoEvent)
+        
+        let shouldSendExpectation = expectation(description: "send event is called")
+        apiClient.onExecute = { request in
+            if let analyticsRequest = request as? AnalyticsRequest {
+                XCTAssertEqual(analyticsRequest.infos.first?.id, infoEvent.id)
```

Comment:
This block requires some cognitive effort, a small suggestion would be

```swift
let expectedID = infoEvent.id
....
apiClient.onExecute = { request in
    if let analyticsRequest = request as? AnalyticsRequest, 
        let capturedId = analyticsRequest.infos.first?.id {

            XCTAssertEqual(capturedId, expectedId, "Expected to send event with id=\(expectedId), got \(capturedId) instead")
            shouldSendExpectation.fulfill()
    }
}
``` 
It can be very helpful to add meaningful assertion messages as you are killing two birds with one stone: make real assertion clear right away and make this section more readable for someone who uses tests as a short description of business requirements.